frontend/src/components/projectEdit/partnersForm.js

```diff
@@ -0,0 +1,278 @@
+import { fetchLocalJSONAPI, pushToLocalJSONAPI } from '../../network/genericJSONRequest';
+import { Listing } from './partnersListing';
+
+export const DateCustomInput = forwardRef(
+  ({ value, onClick, date, handleClear, isStartDate = true, hideCloseIcon = false }, ref) => {
+    return (
+      <div className="relative">
+        <FormattedMessage {...messages[isStartDate ? 'partnerStartDate' : 'partnerEndDate']}>
+          {(message) => {
+            return (
+              <input
+                className="ba b--grey-light pa2 pointer"
+                onClick={onClick}
+                ref={ref}
+                value={value}
+                style={{ height: '42px', boxSizing: 'border-box', maxWidth: '9rem' }}
+                placeholder={message}
+              />
+            );
+          }}
+        </FormattedMessage>
+
+        {(date && hideCloseIcon) || !date ? (
+          <div
+            className="absolute right-1 pointer"
+            style={{ top: '0.9rem' }}
+            onClick={onClick}
+            role="button"
+          >
+            <ChevronDownIcon style={{ color: 'grey', width: '12px', height: '12px' }} />
+          </div>
+        ) : null}
+
+        {date && !hideCloseIcon ? (
+          <div
+            className="absolute right-1 pointer"
+            style={{ top: '0.75rem' }}
+            onClick={handleClear}
+            role="button"
+          >
+            <CloseIcon style={{ color: 'grey', width: '10px', height: '10px' }} />
+          </div>
+        ) : null}
+      </div>
+    );
+  },
+);
+
+DateCustomInput.propTypes = {
+  value: PropTypes.string,
+  date: PropTypes.instanceOf(Date),
+  onClick: PropTypes.func.isRequired,
+  handleClear: PropTypes.func,
+  isStartDate: PropTypes.bool,
+  hideCloseIcon: PropTypes.bool,
+};
+
+export const PartnersForm = () => {
+  const [selectedPartner, setSelectedPartner] = useState({});
+  const [dateRange, setDateRange] = useState({
+    startDate: new Date(),
+    endDate: null,
+  });
+  const [errorMessage, setErrorMessage] = useState({});
+  const userDetails = useSelector((state) => state.auth.userDetails);
+  const token = useSelector((state) => state.auth.token);
+  const queryClient = useQueryClient();
+  const { id } = useParams();
+
+  useEffect(() => {
+    if (
+      selectedPartner &&
+      errorMessage.id &&
+      errorMessage.id === messages.partnerNotSelectedError.id
+    ) {
+      setErrorMessage({});
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedPartner]);
+
+  useEffect(() => {
+    if (!dateRange.endDate && errorMessage.id === messages.partnerEndDateError.id) {
+      setErrorMessage({});
+      return;
+    }
+
+    if (
+      dateRange.endDate &&
+      dateRange.startDate < dateRange.endDate &&
+      errorMessage.id === messages.partnerEndDateError.id
+    ) {
+      setErrorMessage({});
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [dateRange]);
+
+  const {
```

Comment:
can you abstract this to `api/projects.js` folder as `useAllPartnersQuery` function and use that instead.
doing this has following advantages:
- makes it look more clean on the component
- helps to reuse the function if needed in other sections of the project